exasol/toolbox/util/dependencies/poetry_dependencies.py

```diff
@@ -85,9 +85,9 @@ class PoetryDependencies(BaseModel):
 
     @staticmethod
     def _extract_from_line(line: str) -> Package:
-        pattern = r"\s+(\d+(?:\.\d+)*)\s+"
+        pattern = r"\s+(\d+(?:\.\S+)*)"
         match = re.split(pattern, line)
-        return Package(name=match[0], version=match[1])  #
```

Comment:
The regex changes ^ resolve the observed issue & I tested it with:

```python
@pytest.mark.parametrize(
    "path",
    [
        pytest.param(path, id=path.name)
        for path in Path("/home/<user>>/Code/").iterdir()
        if path.name
        not in {
            ".idea",
            "0_docker_vm_setup",
            "exasol-driver-go",
            "project-keeper",
            "project-overview",
            "test",
            "websocket-api",
        }
    ],
)
def test_dependency_license(path):
    poetry_dep = PoetryToml.load_from_toml(working_directory=path)
    PoetryDependencies(
        groups=poetry_dep.groups, working_directory=path
    ).direct_dependencies
```

Reply:
That being said though, do we want to change anything else? We assume a general pattern, but if the pattern isn't met, would we want dependency:licenses to raise an error or silently warn about it?

Reply:
Or maybe there is a better source for the extraction operation?
i.e. https://github.com/pypa/packaging/blob/main/src/packaging/version.py#L185

Reply:
As a first step, I personally think silently warn is the better option. 

You can use `split(maxsplit=2)` to only split at the first 2 spaces.

Regarding versions, I tend to be more strict and to limit freedom.
(as always): Compare project-keeper's [ExasolVersionMatcher.java](https://github.com/exasol/project-keeper/blob/main/shared-model-classes/src/main/java/com/exasol/projectkeeper/shared/ExasolVersionMatcher.java#L11)

Reply:
But when this is about general versions in the python Universe, then we need to adapt to the variants used out there.